tests/test_connection_options.py

```diff
@@ -30,6 +35,24 @@ def test_postgres_explicit_ssl():
     assert kwargs == {"ssl": True}
 
 
+urls_with_options = [
+    (f"{POSTGRES_URLS[0]}?min_size=1&max_size=20", suppress()),
+    (f"{POSTGRES_URLS[0]}?min_size=0&max_size=0", pytest.raises(ValueError)),
+    (f"{POSTGRES_URLS[0]}?min_size=10&max_size=0", pytest.raises(ValueError)),
+]
+
+
+@pytest.mark.parametrize("database_url, expectation", urls_with_options)
+@async_adapter
+async def test_postgres_pool_size_connect(database_url, expectation):
```

Comment:
I think we could do with structuring this a bit more simply. Not sure exactly how, but I'd rather have something a bit more plain.

The empty `supress` is used in a non-obvious way, and `POSTGRES_URLS` is defined at a long distance from the place it's used.

Reply:
I moved POSTGRES_URLS where it's used.
as for supress we can maybe use a more explicit does_not_raise() that is either its own contextmanager (https://github.com/encode/databases/pull/129/commits/c6ab59ed6a8ecb3f66875f959e7f18e9a6f8e452)
either directory an alias for ExitStack (https://github.com/encode/databases/pull/129/commits/3fd7fbac1525ca6fa592ce27913c7e0882a418b7)